packages/block-editor/src/components/provider/use-block-entity-sync.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { last, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { useEffect, useRef } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+export function useBlockEntitySync( {
+	registry,
```

Comment:
Do we really need this here, we can retrieve it from the context right (useContext)?

Reply:
`useRegistry`, yeah.

Reply:
Also the name `useBlockEntitySync`. the block editor doesn't know about "entities", could just be `useBlockSync` or something like that.

Reply:
Both good ideas :) will do